frontend/src/pages/FindChain.tsx

```diff
@@ -42,27 +42,24 @@ const FindChain = () => {
 
   const classes = makeStyles(theme as any)();
 
+  const chains = useContext(ChainsContext);
+
   const [viewport, setViewport] = useState<IViewPort | {}>({});
-  const [chainData, setChainData] = useState<IChain[]>([]);
+  const chainData = chains.filter(({ published }) => published);
   const [selectedChain, setSelectedChain] = useState<IChain | null>(null);
   const [showPopup, setShowPopup] = useState(false);
-  const [filteredChains, setFilteredChains] = useState<IChain[]>([]);
+  const [filteredChains, setFilteredChains] = useState<IChain[]>(chainData);
   const [userId, setUserId] = useState("");
   const [role, setRole] = useState<string | null>(null);
 
+  useEffect(() => {
+    setFilteredChains(chainData);
+  }, [chains]);
```

Comment:
Is this still necessary, you already have `useState<IChain[]>(chainData);`

Reply:
On FindChain.tsx first render chainData is most likely the default Context value of [] as the async call at ChainsProvider is not finished yet. After the async call is finished then we need to update the state and hence need to use useEffect

Reply:
You might a bit of an ordering issue then. Because it looks like the searchbar can change things, which will again be overridden by the searchbar component. So if someone searches before the data is loaded..

This whole ping-ponging the data between components might get a little hard to debug if things go wrong. I might be good to think of a way to structure things a little more simply. Maybe instead of pushing the chain data back and forwards, it might be good to expose a filter callback property on SearchBar? That way you keep the chain data in one spot.

I'll approve for now, but might be worth thinking about. Code that's too clever/complicated has a way of causing some very annoying bugs and maintenance issues.

Reply:
I'll address and refactor the search bar in separate PR so this PR kept focused on React Context